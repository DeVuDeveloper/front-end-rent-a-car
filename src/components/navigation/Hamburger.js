/* eslint-disable arrow-body-style */
/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { MenuIcon, XIcon } from '@heroicons/react/solid';
import logo from '../../assets/images/logo.png';
import twitter from '../../assets/images/twitter-icon.png';
import facebook from '../../assets/images/facebook-icon.png';
import linkedin from '../../assets/images/linkedin-icon.png';
import github from '../../assets/images/github-icon.png';
import Logout from '../auth/Logout';
import './hamburger.css';

function classNames(...classes) {
  return classes.filter(Boolean).join(' ');
}

const menuItems = [
  { name: 'Cars', path: '/home' },
  { name: 'Reserve', path: '/reserve' },
  { name: 'My reservations', path: '/reservations' },
];

const social = [
  { icon: twitter },
  { icon: facebook },
  { icon: linkedin },
  { icon: github },
];

const Hamburger = ({ currentUser }) => {
  const [mobileMenu, setMobileMenu] = useState(false);
  return (
    <header
      className={`flex flex-col px-8  ${
        mobileMenu ? 'h-screen header' : 'h-fit'
      } fixed top-0 w-screen shadow-xl md:hidden`}
    >
      <a href="/" className="text-lg text-slate-50">
        <img
          src={logo}
          className={`w-24 ${mobileMenu ? 'hidden' : 'block'}`}
          alt="logo"
        />
      </a>
      <nav>
        <div className="text-slate-50 mobile-btns">
          <button
            type="button"
            className={` text-lime-500 ${!mobileMenu ? 'block' : 'hidden'}`}
            onClick={() => setMobileMenu(true)}
          >
            <MenuIcon className="h-8" />
          </button>
          <button
            type="button"
            className={classNames(
              mobileMenu ? 'block' : 'hidden',
              'text-slate-900',
            )}
            onClick={() => setMobileMenu(false)}
          >
            <XIcon className="h-8" />
          </button>
        </div>
        <div className={classNames(mobileMenu ? 'block' : 'hidden')}>
          <ul className="flex flex-col text-slate-900 font-bold text-lg gap-8 items-center mt-44">
            {menuItems.map((item) => (
              <li key={item.name}>
                <a href={item.path}>{item.name}</a>
              </li>
            ))}
          </ul>
          <div className="hamburger-column">
            {currentUser.role === 'admin' && (
              <NavLink
                to="/add_car"
                className={({ isActive }) => (isActive ? 'active' : 'inactive')}
              >
                <p>ADD CAR</p>
              </NavLink>
            )}

            {currentUser.role === 'admin' && (
              <NavLink
                to="/delete"
                className={({ isActive }) => (isActive ? 'active' : 'inactive')}
              >
                <p>DELETE CAR</p>
              </NavLink>
            )}

            <span className="logout">
              <Logout />
            </span>
          </div>
        </div>
      </nav>
      <div
        className={classNames(
          mobileMenu ? 'block' : 'hidden',
          'flex flex-col items-center mt-32',
        )}
      >
        <div className="flex gap-2">
          {social.map((item) => (
            <a href="/" key={item.icon}>
              <img src={item.icon} alt="social-icon" className="w-8" />
            </a>
          ))}
        </div>
        <p className="text-gray-700 font-bold mt-2">Â© 2022</p>
      </div>
    </header>
  );
};

const mapStateToProps = ({ auth: { currentUser } }) => {
  return { currentUser };
};

export default connect(mapStateToProps)(Hamburger);
