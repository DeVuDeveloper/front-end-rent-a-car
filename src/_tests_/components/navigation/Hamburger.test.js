import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Hamburger from '../../../components/navigation/Hamburger';
import store from '../../../redux/configureStore';

it('should render Sidebar correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Hamburger />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
