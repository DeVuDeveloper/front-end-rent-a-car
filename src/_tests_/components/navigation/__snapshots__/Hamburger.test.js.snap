// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar correctly 1`] = `
<header
  className="flex flex-col px-8  h-fit fixed top-0 w-screen shadow-xl md:hidden"
>
  <a
    className="text-lg text-slate-50"
    href="/"
  >
    <img
      alt="logo"
      className="w-24 block"
      src="logo.png"
    />
  </a>
  <nav>
    <div
      className="text-slate-50 mobile-btns"
    >
      <button
        className=" text-lime-500 block"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="h-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <button
        className="hidden text-slate-900"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="h-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      className="hidden"
    >
      <ul
        className="flex flex-col text-slate-900 font-bold text-lg gap-8 items-center mt-44"
      >
        <li>
          <a
            href="/home"
          >
            Cars
          </a>
        </li>
        <li>
          <a
            href="/reserve"
          >
            Reserve
          </a>
        </li>
        <li>
          <a
            href="/reservations"
          >
            My reservations
          </a>
        </li>
        <li>
          <a
            href="/"
          >
            Add car
          </a>
        </li>
        <li>
          <a
            href="/delete"
          >
            Delete car
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="hidden flex flex-col items-center mt-32"
  >
    <div
      className="flex gap-2"
    >
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-8"
          src="twitter-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-8"
          src="facebook-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-8"
          src="linkedin-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-8"
          src="github-icon.png"
        />
      </a>
    </div>
    <p
      className="text-gray-700 font-bold mt-2"
    >
      Â© 2022
    </p>
  </div>
</header>
`;
